back-end/app/Http/Controllers/Lesson/LessonsController.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+namespace App\Http\Controllers\Lesson;
+
+use App\Models\Lesson;
+use Illuminate\Http\Request;
+use App\Http\Traits\AdminTrait;
+use App\Http\Controllers\Controller;
+use Illuminate\Support\Facades\Gate;
+use Illuminate\Auth\Access\AuthorizationException;
+
+
+class LessonsController extends Controller
+{
+    // Call traits for reusable codes for checking if the user is admin
+    use AdminTrait;
+
+    // Display a listing of the Lessons
+    public function index()
+    {
+        $lessons = Lesson::all();
+
+        return response()->json([
+            'data' => $lessons
+        ]);
+    }
+
+    // Store a newly created Lesson in storage.
+    public function store(Request $request, $lesson_id)
+    {
+        $this->isAdmin();
+
+        $data = $request->validate([
+            'title' => 'required|string|max:191',
+            'description' => 'required|string|max:191',
+        ]);
+
+        Lesson::findOrFail($lesson_id)->update([
+            'title' => $data['title'],
+            'description' => $data['description'],
+        ]);
```

Comment:
This is for update. While your update method has create logic. Swap it nalang.

Reply:
Oh, I'm sorry, it got switched with the update method. I'll fix all of the comments and re-push.